### PR TITLE
FEAT: Start to work on the responsive layout

### PR DESCRIPTION
**Pull request by ips-coding-challenge**

### CODE REVIEW

File: src/components/cards/Item.tsx [hunk 1]
```diff
@@ -137,16 +137,12 @@ const Item: React.FC<PropTypes> = ({ data, category, history }) => {
...
                 onClick={addItemToShopList}
-                style={{
-                    height: 'fit-content',
-                    width: 'fit-content',
-                }}
             >
-                <MdAdd size={24} />
-            </Button>
+                <MdAdd className="text-gray-light" size={24} />
+            </div>
```
Comment (fajfara) [inline src/components/cards/Item.tsx:145]: Why did you change this to a div?

File: src/components/cards/List.tsx [hunk 2]
```diff
@@ -56,13 +56,13 @@ const List = ({ list: { id, name, status, created_at } }: ListProps) => {
...
+                <div className="font-bold mb-2 md:mb-0">{name}</div>
 
-                <div className="flex items-center w-1/2 justify-around">
+                <div className="flex flex-col md:flex-row items-center md:w-1/2 justify-around">
                     {/* Icon + Date */}
-                    <div className="flex items-center">
+                    <div className="flex items-center mb-3 md:mb-0">
                         <MdDateRange
                             style={iconStyle}
                             className="text-gray-light mr-2"
```
Comment (fajfara) [inline src/components/cards/List.tsx:68]: On screens smaller than 768px, aligning content here on the center looks kind of weird to me, especially since we have content to the left (navbar) and most the text is left-aligned. Also on tablet screens should we display each list as 50% width? That way we can fit more content on tablets, but on mobile, we just use the full width.

File: src/components/cards/Item.tsx (hunk 1 above)
Comment (ips-coding-challenge) [reply src/components/cards/Item.tsx:145]: Because of the default classes in the button which add extra margin/padding that I couldn't control directly in this component.

Comment (fajfara) [reply src/components/cards/Item.tsx:145]: Can we adjust the button component for that? For example, we remove spacing from default classes and add them to props. So now if the spacing is in props use those classes otherwise we just use some default spacing classes.

Comment (ips-coding-challenge) [reply src/components/cards/Item.tsx:145]: As this button has not the same design as the "standard" buttons, it is not better to modify only that button?

File: src/components/cards/List.tsx (hunk 2 above)
Comment (ips-coding-challenge) [reply src/components/cards/List.tsx:68]: I'm not sure to understand. On smaller screens, I can't keep the original List design as I have not enough space. And I tried to align everything to the left ( with the flex column ) and It feels kinda weird I think. What do you mean by displaying each list as 50% width? Do you mean to have 2 columns of lists?

Comment (fajfara) [reply src/components/cards/List.tsx:68]: Yeah, what I meant for the list display on the tablet was to put it into two columns. I guess we can wait with that and see how it looks with a lot of lists (if we leave too much-unused whitespace on the right). Regarding the center-aligned cards, I still think it breaks the flow too much if the content is centered
Screenshot at Sep 16 11-55-07
For stuff like this i usually just like to follow something like material design guidelines, in our example this would me material design cards.

File: src/components/cards/Item.tsx (hunk 1 above)
Comment (fajfara) [reply src/components/cards/Item.tsx:145]: I suppose. But still, change it to button element instead of div.

File: src/components/cards/List.tsx (hunk 2 above)
Comment (ips-coding-challenge) [reply src/components/cards/List.tsx:68]: I will check the material design cards ;)

Ok to wait for the lists in two columns because I'm not sure of the user experience. Anyway, It was just a first pass, it's not a lot of work to modify things if we change our mind, or just to do some design tests if we want ;)

**Review by fajfara — APPROVED** [review]
